core/hardhat.config.ts

```diff
@@ -5,11 +5,23 @@ import ethers from '@nomiclabs/hardhat-ethers';
 dotenv.config();
 
 const LOCAL_RPC_URL = process.env.LOCAL_RPC_URL || 'http://localhost:8545';
+const REMOTE_RPC_URL = process.env.REMOTE_RPC_URL;
+
+if (!REMOTE_RPC_URL) {
+    throw new Error('REMOTE_RPC_URL env variable not set, please check your `hardhat/.env` file');
+}
 
 const customNetworkConfig = {
     testnetwork: {
         url: LOCAL_RPC_URL,
     },
```

Comment:
Why do you need to create this `testnetwork` and provide url to the hardhat itself within the hardhat config? (also, this env var is not declared in the dockerfile)

Reply:
essentialy this is related to the ci cd.

When adding it i did not discover how to point to the hardhat fork that is spawned by default. So instead i created a separate instance that has an adderess that we know. This fork is the `testnetwork`.

> also, this env var is not declared in the dockerfile)

[false](https://github.com/sidestream-tech/unified-auctions-ui/pull/426/files#diff-e95472413b240ae7729b7bea4f74662574ce035d769cf78f1191e47a277d79cbL21).

Reply:
https://hardhat.org/hardhat-network/docs/overview#how-does-it-work?

> It runs as either an in-process or stand-alone daemon, servicing JSON-RPC and WebSocket requests.

- currently it's a standalone because in this case we can configure the address it is at.
- when we do `npm run hardhat` the daemon starts and is located at localhost, so when developing we need this value to be `localhost:8545`
- when we are on ci pipeline this hardhat node is deployed as a separate service and thus we have to configure the env variable so that it points to the service in the drone. 
- The network is used for tests (see `package.json`) because otherwise the `hardhat test` will spawn its own instance of hardhat that apparently doesn't have an url to reach it through (from what i remember when implementing it some months ago)

TL;DR: we need this env var so that the tests know what address the node is at.

Reply:
> TL;DR: we need this env var so that the tests know what address the node is at.

Yes, exactly, I was not asking why do we need this variable. We need this variable in order to point tests toward service that runs RPC endpoint, that is correct. I were asking about the `testnetwork` config. 

> When adding it i did not discover how to point to the hardhat fork that is spawned by default. So instead i created a separate instance that has an adderess that we know. This fork is the `testnetwork`.

This was very confusing answer. After reading the documentation, I can finally understand what is happening:

- When we run `npx hardhat node`, the `hardhat` portion of the config is used and this creates a new RPC endpoint (in other words a new node is created for others to run against)
- When we run `npx hardhat --network testnetwork run ./simulations/index.ts`, the `testnetwork` portion of the config is created, but it doesn't create new RPC endpoint, it connects to the provided one and executes provided script over it

Since those two things will be run in two different processes, this now make more sense, but doesn't make it less confusing. Not sure if it was a good decision to merge those two folders together. Maybe it will make sense to at least store configs run by separate processes separately (eg, currently you [validate REMOTE_RPC_URL](https://github.com/sidestream-tech/unified-auctions-ui/blob/e683d4a085633e81414831523c2b89b210f2264a/core/hardhat.config.ts#L10-L12) for both processes, while you need it only in one)? `npx hardhat` have `--config` option to achieve separation into `remote.hardhat.config.ts` and `local.hardhat.config.ts`

Reply:
i don't think that "removing" the default option (`hardhat.config.ts`) is an improvement. How about assuming that the remote option is the default and the local has to use different config? (corresponding change applied.)

Reply:
> i don't think

Can you elaborate why? My suggestion was about making it less confusing through consistency of the terms `remote`/`local`, which your suggestion contradicts without explaining the reasoning

Reply:
the reasoning is that the absence of the default config is confusing as well - i would expect a config that provides a valid fork / node to be the default option and the other configs to be the ones that cover specific cases.

on top of that, we would have to write out the config file arg for every single command `npx hardhat` - that is inconvenient.  It's possible to save some keystrokes there.

Also, if i think about naming then distinction between `local` and `remote` is not exactly clear as well:

- the `local` as we call it now is just there to point to the local node via `url` or in other words it shows "which node to use"
- the `remote` is there to configure on how we should create fork (`forking/url`) and does not provide the `url` parameter.

So if we stick to double-naming then i would say it should be `local.hardhat.config` and `fork.hardhat.config`